db/merge_helper.cc

```diff
@@ -153,12 +153,13 @@ Status MergeHelper::MergeUntil(InternalIterator* iter,
     ParsedInternalKey ikey;
     assert(keys_.size() == merge_context_.GetNumOperands());
 
-    if (ParseInternalKey(iter->key(), &ikey) != Status::OK()) {
+    Status pikStatus = ParseInternalKey(iter->key(), &ikey);
+    if (pikStatus != Status::OK()) {
       // stop at corrupted key
       if (assert_valid_internal_key_) {
         assert(!"Corrupted internal key not expected.");
-        s = Status::Corruption("Corrupted internal key not expected.");
-        return s;
+        return Status::Corruption("Corrupted internal key not expected. ",
+                                  pikStatus.getState());
```

Comment:
Not sure what to do about this one but this is a very odd message.   Like there is a time when a Corrupted internal key *is* expected?